### DISCUSSION TITLE
Add `relative_timestamp` sensor device class

### DISCUSSION THREAD

**Discussion by raman325** [home-assistant/architecture#890] [Entity Models]
## Background

In the [developer docs](https://developers.home-assistant.io/docs/core/entity/sensor), we say that `DURATION` should not be used for values that update simply due to time passing. There are, however, valid use cases where it may be useful to know a relative duration to now, like system/service/application uptime, time until next transit arrival, etc. To accommodate this, I propose a new device class that takes a `datetime` as an input, so from a backend perspective, this is no different than the `TIMESTAMP` device class. The frontend, however, could use this new device class as a trigger to display a constantly updating relative duration to now.

## Proposal

1. Add a new `SensorDeviceClass` enum called `RELATIVE_TIMESTAMP`. If no timezone is provided, the HA local timezone is inserted
2. Add frontend logic that renders a duration value = (the sensor's state - now())

## Alternatives To Consider

- Build a card that does this for `TIMESTAMP` sensors
- Add some sort of configuration option in the UI that allows the user to switch the display from date/time to relative duration, possibly even with a configurable date/time to compare it to (options would be `now` or a static date/time value)

**Comment by frenck** [comment]
The current `timestamp` device class, already shows a relative time in the frontend? So, if I understand your suggestion right, it would mean the only difference is the fact it updates live?

Wouldn't it make more sense to just make the existing relative time for the `timestamp` device class to update live in the frontend?

../Frenck

**Reply by frenck** [reply]
Nevermind, just learned from bram it is actually updating already :)

**Reply by raman325** [reply]
whoops! guess this is unneeded then. See my comment below

**Comment by bramkragten** [comment]
You can change the format of a timestamp entity in the frontend now: https://www.home-assistant.io/dashboards/entities/#format
This should maybe be implemented in more places, and made more consistent...
Relative times are updated every 60 seconds btw :-)

![CleanShot 2023-05-08 at 10 53 48](https://user-images.githubusercontent.com/5662298/236781200-92488474-60ac-4656-80bb-289c3a98e632.png)

**Reply by elupus** [reply]
Its semantically different thou. One thing is a point in time, while the other is a count down (or up). For a count down/up the integration might want to have a say in presentation format defaults. Like always hours, always seconds.

**Reply by raman325** [reply]
ha I was not aware of this. To @elupus 's point though, maybe a countdown format would be a good addition? As a start it could provide defaults, and we could extend it later to either allow the integration or the user to have a say in the presentation format

**Reply by raman325** [reply]
hmm I'm actually not sure if they are different. Relative seems to be OK. I don't know if a 60 second refresh will work in every use case but it will work in many
